site.go

```diff
@@ -123,6 +161,41 @@ func (s *Site) IsValid() error {
 		}
 	}
 
+	if s.ResizingService != "imgix" && s.ResizingService != "thumbor" && s.ResizingService != "thumbor+cloudfront" && s.ResizingService != "" {
+		return errors.New(fmt.Sprintf("Unrecognized/Unimplemented resizing service '%s',"+
+			" valid options are imgix, thumbor, thumbor+cloudfront", s.ResizingService))
+	}
+
+	if s.ResizingService == "thumbor" && s.ResizingServiceSecret == "" {
+		return errors.New("Thumbor resizing service requires use of a shared secret for URL signing")
+	}
+
+	if s.ResizingService == "thumbor+cloudfront" && (s.AWS_CLOUDFRONT_PRIVATE_KEY_PATH == "" || s.AWS_CLOUDFRONT_PRIVATE_KEY_PAIR_ID == "") {
+		return errors.New("thumbor+cloudfront resizing service requires you to provision a private key " +
+			"and provide the path to the private key(config AWSCloudfrontKeyPath)," +
+			" along with the associated key pair id (config AWSCloudfrontKeyPairId)")
+	} else if s.ResizingService == "thumbor+cloudfront" && s.AWS_CLOUDFRONT_PRIVATE_KEY_PATH != "" {
+		// borrowed from: https://github.com/ianmcmahon/encoding_ssh
```

Comment:
Is the key file being loaded in two places? Here and init?

Maybe refactor this out into it's own function that sets the key file as well, and call that after we load our config values but before calling `IsValid()`. Then `IsValid()` just ensures that the required values are present.

Reply:
so thought it'd make sense to validate the file itself, rather than fail once trying to use it - but i realise that it's better to just do a full load twice and move this in to a method as you say... didn't see the forrest for the trees, was trying to avoid parsing the file without avoiding parsing the file. will change.